File: Sesion 7/Actividades/Actividad 3/Actividad 3 S7.py
class Cliente:
    def __init__(self, nombre, apellido, correo, direccion, numero_tarjeta):
        self._nombre = nombre
        self._apellido = apellido
        self._correo = correo
        self._direccion = direccion
        self._numero_tarjeta = numero_tarjeta

    def obtener_informacion(self):
        info = f"Nombre: {self._nombre}\n"
        info += f"Apellido: {self._apellido}\n"
        info += f"Correo: {self._correo}\n"
        info += f"Direccion: {self._direccion}\n"
        info += f"Numero de tarjeta: {self._numero_tarjeta}\n"
        return info

    def encriptar_informacion(self):
        encriptacion = "".join(str(ord(c)) for c in self._correo)
        return encriptacion


class ClienteSeguro(Cliente):
    def __init__(self, nombre, apellido, correo, direccion, numero_tarjeta, codigo_seguridad):
        super().__init__(nombre, apellido, correo, direccion, numero_tarjeta)
        self._codigo_seguridad = codigo_seguridad

    def verificar_autenticidad(self, correo_ingresado, numero_tarjeta_ingresado):
        return correo_ingresado == self._correo and numero_tarjeta_ingresado == self._numero_tarjeta

    def obtener_informacion_encriptada(self):
        return self.encriptar_informacion()


cliente_seguro = ClienteSeguro("goku", "sayayin 1000", "gokulegana@hotmail_gmail_twitter.com", "Bajo un Puente", "1234 5678 9012 3456", "ABC123")
print("Informacion del cliente:")
print(cliente_seguro.obtener_informacion())
print("Informacion encriptada:", cliente_seguro.obtener_informacion_encriptada())
autenticado = cliente_seguro.verificar_autenticidad("gokulegana@hotmail_gmail_twitter.com", "1234 5678 9012 3456")
print("Cliente autenticado:", "Si" if autenticado else "No")

File: Sesion 7/Actividades/Actividad 3/Actividad 3 S7.java
class Cliente {
    protected String nombre;
    protected String apellido;
    protected String correo;
    protected String direccion;
    protected String numeroTarjeta;

    public Cliente(String nombre, String apellido, String correo, String direccion, String numeroTarjeta) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.correo = correo;
        this.direccion = direccion;
        this.numeroTarjeta = numeroTarjeta;
    }

    public String obtenerInformacion() {
        StringBuilder info = new StringBuilder();
        info.append("Nombre: ").append(nombre).append("\n");
        info.append("Apellido: ").append(apellido).append("\n");
        info.append("Correo: ").append(correo).append("\n");
        info.append("Direccion: ").append(direccion).append("\n");
        info.append("Numero de tarjeta: ").append(numeroTarjeta).append("\n");
        return info.toString();
    }

    protected String encriptarInformacion() {
        StringBuilder encriptacion = new StringBuilder();
        for (char c : correo.toCharArray()) {
            encriptacion.append((int) c);
        }
        return encriptacion.toString();
    }
}

class ClienteSeguro extends Cliente {
    private String codigoSeguridad;

    public ClienteSeguro(String nombre, String apellido, String correo, String direccion, String numeroTarjeta, String codigoSeguridad) {
        super(nombre, apellido, correo, direccion, numeroTarjeta);
        this.codigoSeguridad = codigoSeguridad;
    }

    public boolean verificarAutenticidad(String correoIngresado, String numeroTarjetaIngresado) {
        return correoIngresado.equals(correo) && numeroTarjetaIngresado.equals(numeroTarjeta);
    }

    public String obtenerInformacionEncriptada() {
        return encriptarInformacion();
    }
}

public class Main {
    public static void main(String[] args) {
        ClienteSeguro clienteSeguro = new ClienteSeguro("goku", "sayayin 1000", "gokulegana@hotmail_gmail_twitter.com", "Bajo un Puente", "1234 5678 9012 3456", "ABC123");
        System.out.println("Informacion del cliente:");
        System.out.println(clienteSeguro.obtenerInformacion());
        System.out.println("Informacion encriptada: " + clienteSeguro.obtenerInformacionEncriptada());
        boolean autenticado = clienteSeguro.verificarAutenticidad("gokulegana@hotmail_gmail_twitter.com", "1234 5678 9012 3456");
        System.out.println("Cliente autenticado: " + (autenticado ? "Si" : "No"));
    }
}